### ISSUE TITLE
Bug in FormatParameterMismatch cop?

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#2110]
@edmz 

With the new version 0.33.0, I am seeing FormatParamterMismatch complain about correct usage of `String#%` (which I happen to prefer over `format`):

My test.rb
```rb
# encoding: utf-8
puts '%s' % 42
```

My RucoCop output:
```
Inspecting 1 file
W

Offenses:

test.rb:2:11: W: Number arguments (0) to String#% mismatches expected fields (1).
puts '%s' % 42
          ^
test.rb:2:11: C: Favor format over String#%.
puts '%s' % 42
          ^

1 file inspected, 2 offenses detected
```

I know how to configure my way out of “Favor `format` over `String#%`”, but the first message looks like a bug.

**Comment by bbatsov** [comment]
Yep, that's definitely a bug.

**Comment by edmz** [comment]
I created PR #2112 that should fix this.
